src/praisonai/praisonai/integrations/managed_local.py

```
@@ -547,7 +627,21 @@ def _persist_message(self, role: str, content: str) -> None:
-        agent = self._ensure_agent()
+        # Note: This method is kept for backwards compatibility but 
+        # cannot provision compute instances. Use execute() instead.
+        if self._inner_agent is None:
+            # Try sync fallback for packages without compute
+            try:
+                import asyncio
+                loop = asyncio.new_event_loop()
+                asyncio.set_event_loop(loop)
+                agent = loop.run_until_complete(self._ensure_agent())
+                loop.close()
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)